### DISCUSSION TITLE
Is it possible to get the velocity of each turbine in a wind farm using Floris ?

### DISCUSSION THREAD

**Discussion by Ooussamaa** [NatLabRockies/floris#640] [Ideas]
Hey guys,

I am wondering if I could get the velocity value on each one of my turbines, as I can do for turbines power using the command "powers = fi.get_turbine_powers()". 
Floris has an attribute ( fi.get_turbine_velocities() ), so how can I proceed please  ?
Thank you 
Happy coding !
Ouuss

**Comment by paulf81** [comment]
Thank you for your comment @Ooussamaa , if you would like that function to be added, you can add an issue requesting it.  There are however a few options presently:

```
from floris.tools import FlorisInterface

# Load the 3-turbine example farm
fi = FlorisInterface("inputs/gch.yaml")
fi.calculate_wake()


# Show all calculated values
print('All points')
print(fi.floris.flow_field.u)

# Show the "effective" velocity at each turbine
print('Effective velocity')
print(fi.turbine_effective_velocities)
```

The first print statement shows the u-value at each point.  For each turbine, using the 3x3 grid, 9 points are computed and the middle one would be the one in the center of the rotor:

```
All points
[[[[[7.59695563 8.         8.29335086]
    [7.59695563 8.         8.29335086]
    [7.59695563 8.         8.29335086]]

   [[5.2996384  4.91633716 5.78544391]
    [4.66865652 4.06937734 5.09662139]
    [5.29964956 4.91635214 5.78545609]]

   [[5.2786327  5.15933197 5.76251267]
    [4.92491367 4.72502107 5.37636904]
    [5.32121864 5.2125903  5.80900236]]]]]
```

Then the "effective" velocity  is the value used in the calculation of power for each turbine:

```
Effective velocity
[[[7.97363299 5.14282155 5.3059964 ]]]
```

Hope this helps!

Paul

**Comment by Ooussamaa** [comment]
Thank you paulf81 for your answer, it wil be for sure a great help for me.
Oussama

**Reply by paulf81** [reply]
My pleasure!
